src/vivarium_census_prl_synth_pop/results_processing/addresses.py

```diff
@@ -88,9 +101,14 @@ def get_zipcode_map(
 
     """
     zip_map_dict = {}
-    output_cols = [column_name, "state_id", "puma"]  # columns in the output we use to map
+    if column_name == "address_id":
+        output_cols = [column_name, "state_id", "puma"]  # columns in the output we use to map
+        groupby_cols = ["state_id", "puma"]
+    else:
+        output_cols = [column_name, "employer_state_id", "employer_puma"]
+        groupby_cols = ["employer_state_id", "employer_puma"]
     simulation_addresses = (
-        obs_data.reset_index()[output_cols].drop_duplicates().set_index("address_id")
+        obs_data.reset_index()[output_cols].drop_duplicates().set_index(column_name)
```

Comment:
Didn't we already drop all the duplicates when we formatted `obs_data`? Do we actually still have duplicates at this point? The same question applies in all the other places we do this.

Reply:
Yes This one should not be here.  We had added it for zipcode map because state and puma are not unique.  I will remove the unnecessary ones.

Reply:
Why would there still be duplicates here? Address id should already be unique on it's own, so the combination of address id, puma, and state should definitely be unique. Am I missing something?

Reply:
I think it's not needed actually. I think I originally added the drop_duplicates, but it *should* be unnecessary assuming we've got address_id without duplicates (which should be the case). I think I may have added it as CYA not knowing how we'd build up the column superset for the address_id columns.